test/WebSites/ModelBindingWebSite/Controllers/FileUploadController.cs

```diff
@@ -0,0 +1,111 @@
+using Microsoft.AspNet.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace ModelBindingWebSite.Controllers
+{
+    public class FileUploadController : Controller
+    {
+        public FileDetails UploadSingle(IFormFile file)
+        {
+            if (!ModelState.IsValid)
+            {
+                return null;
+            }
+
+            var reader = new StreamReader(file.OpenReadStream());
+            var fileContent = reader.ReadToEnd();
+            ContentDispositionHeaderValue contentDisposition;
+            var parsedContentDisposition = ContentDispositionHeaderValue.TryParse(file.ContentDisposition,
+                                                                                  out contentDisposition);
+            var fileDetails = new FileDetails
+            {
+                Filename = contentDisposition.FileName,
+                Content = fileContent
+            };
+            return fileDetails;
+        }
+
+        public FileDetails[] UploadMultiple(IEnumerable<IFormFile> files)
+        {
+            if (!ModelState.IsValid)
+            {
+                return null;
+            }
+
+            var fileDetailsList = new List<FileDetails>();
+            foreach (var file in files)
+            {
+                var reader = new StreamReader(file.OpenReadStream());
+                var fileContent = reader.ReadToEnd();
+                ContentDispositionHeaderValue contentDisposition;
+                var parsedContentDisposition = ContentDispositionHeaderValue.TryParse(file.ContentDisposition,
+                                                                                      out contentDisposition);
+                var fileDetails = new FileDetails
+                {
+                    Filename = contentDisposition.FileName,
+                    Content = fileContent
+                };
+                fileDetailsList.Add(fileDetails);
+            }
+            return fileDetailsList.ToArray();
+        }
+
+        public IDictionary<string, IList<FileDetails>> UploadMultipleList(IEnumerable<IFormFile> filelist1,
+                                                                     IEnumerable<IFormFile> filelist2)
+        {
+            if (!ModelState.IsValid)
+            {
+                return null;
+            }
+
+            var fileDetailsDict = new Dictionary<string, IList<FileDetails>>
+            {
+                { "filelist1", new List<FileDetails>() },
+                { "filelist2", new List<FileDetails>() }
+            };
+            var fileDetailsList = new List<FileDetails>();
+            foreach (var file in filelist1.Concat(filelist2))
+            {
+                var reader = new StreamReader(file.OpenReadStream());
+                var fileContent = reader.ReadToEnd();
+                ContentDispositionHeaderValue contentDisposition;
+                var parsedContentDisposition = ContentDispositionHeaderValue.TryParse(file.ContentDisposition,
```

Comment:
don't use TryParse if you're not going to handle the failure case. Just use Parse and let it blow up